File: Course_01/C++ Language/Vetores/dados_pessoais.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main()
{
    int i, n;

    cout <<"Quantas pessoas serao digitadas? ";
    cin >> n;

    double altura[n], menorAltura, maiorAltura, mediaF, somaF(0);
    char sexo[n];
    int conteF(0);

    for (i = 0; i < n; i++) {
        cout <<"Altura da " << i + 1 <<"a pessoa: ";
        cin >> altura[i];
        cout <<"Genero da " << i + 1 <<"a pessoa: ";
        cin >> sexo[i];
    }

    menorAltura = altura[0];
    maiorAltura = altura[0];

    for (i = 0; i < n; i++) {
        if (menorAltura > altura[i]) {
            menorAltura = altura[i];
        }
        if (maiorAltura < altura[i]) {
            maiorAltura = altura[i];
        }
        if (sexo[i] == 'F') {
            conteF++;
            somaF += altura[i];
        }
    }

    mediaF = somaF / conteF;

    cout << "Menor altura = " << menorAltura <<endl;
    cout << "Maior altura = " << maiorAltura <<endl;
    cout << fixed << setprecision(2);
    cout << "Media das alturas das mulheres = " << mediaF <<endl;
    cout << "Numero de homens = " << n - conteF << endl;
    return 0;
}

File: Course_01/Python/Vetores/dados_pessoais.py
n = int(input("Quantas pessoas serão digitadas? "))

altura = [0 for x in range(n)]
genero = [0 for x in range(n)]

for i in range(0, n):
    altura[i] = float(input(f"Altura da {i + 1}a pessoa: "))
    genero[i] = input(f"Genero da {i + 1}a pessoa: ")

menorAltura = maiorAltura = altura[0]
somaAlturasMulheres = contagemMulheres = 0

for i in range(0, n):
    if menorAltura > altura[i]:
        menorAltura = altura[i]
    if maiorAltura < altura[i]:
        maiorAltura = altura[i]
    if genero[i] == "F":
        somaAlturasMulheres += altura[i]
        contagemMulheres += 1
print(f"Menor altura = {menorAltura}")
print(f"Maior altura = {maiorAltura}")
print(f"Média das alturas das mulheres = {(somaAlturasMulheres / contagemMulheres):.2f}")
print(f"Número de homens = {n - contagemMulheres}")